db/document.go

```diff
@@ -527,14 +536,32 @@ func (s *SyncData) GetSyncCas() uint64 {
 	return base.HexCasToUint64(s.Cas)
 }
 
+func HasUserXattrChanged(rawUserXattr []byte, rawUserXattrHash string) bool {
+	// If doc has no user xattr but crc indicates there used to be one trigger import.
+	if len(rawUserXattr) == 0 && rawUserXattrHash != "" {
+		return true
+	}
+
+	// If doc has a user xattr set and crc doesn't match there has been a change. Trigger import
+	if len(rawUserXattr) > 0 && base.Crc32cHashString(rawUserXattr) != rawUserXattrHash {
+		return true
+	}
```

Comment:
Not sure whether this is personal preference, but to me this becomes much simpler as:
```
// If previous hash is empty, return based on presence of new hash
if rawUserXattrHash == "" {
   return len(rawUserXattr) > 0
}
// Otherwise compare previous hash with new hash
return base.Crc32cHashString(rawUserXattr) == rawUserXattrHash
```
There's a bit of inefficiency (calculating the hash when the user xattr is removed), so you could argue for an early return there, but overall I think it's clearer?

Reply:
I personally find mine easier to read. Its more the first bit I'm not a fan of really.

Reply:
Not sure the above will work well actually either. In the case where a hash is present but there is no xattr this will fall through to the hash check which is extra work too.